From 197fc5cef4c5af920c791b383a5c8d6c2341cf26 Mon Sep 17 00:00:00 2001
From: Sven Eckelmann <sven@narfation.org>
Date: Wed, 30 Nov 2016 11:54:46 +0100
Subject: [PATCH 7/7] batman-adv: Add get_link_net compat code for Linux <
 4.0.0

LEDE is still using Linux 3.18 on some systems. Allow to build batman-adv
for it by not checking the netns of a possible netdev link during the
batadv-on-batadv check.

Signed-off-by: Sven Eckelmann <sven@narfation.org>
---
 ...dd-compat-fallback-for-batadv_getlink_net.patch | 29 ++++++++++++++++++++++
 1 file changed, 29 insertions(+)
 create mode 100644 batman-adv/patches/0005-Add-compat-fallback-for-batadv_getlink_net.patch

diff --git a/batman-adv/patches/0005-Add-compat-fallback-for-batadv_getlink_net.patch b/batman-adv/patches/0005-Add-compat-fallback-for-batadv_getlink_net.patch
new file mode 100644
index 0000000..23592e8
--- /dev/null
+++ b/batman-adv/patches/0005-Add-compat-fallback-for-batadv_getlink_net.patch
@@ -0,0 +1,29 @@
+From: Sven Eckelmann <sven@narfation.org>
+Date: Fri, 23 Sep 2016 14:55:38 +0200
+Subject: [PATCH] Add compat fallback for batadv_getlink_net
+---
+ net/batman-adv/hard-interface.c | 4 ++++
+ 1 file changed, 4 insertions(+)
+
+diff --git a/net/batman-adv/hard-interface.c b/net/batman-adv/hard-interface.c
+index 08ce361..e6ea2f9 100644
+--- a/net/batman-adv/hard-interface.c
++++ b/net/batman-adv/hard-interface.c
+@@ -95,6 +95,9 @@ out:
+ static const struct net *batadv_getlink_net(const struct net_device *netdev,
+ 					    const struct net *fallback_net)
+ {
++#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 0, 0)
++	return fallback_net;
++#else
+ 	if (!netdev->rtnl_link_ops)
+ 		return fallback_net;
+ 
+@@ -102,6 +105,7 @@ static const struct net *batadv_getlink_net(const struct net_device *netdev,
+ 		return fallback_net;
+ 
+ 	return netdev->rtnl_link_ops->get_link_net(netdev);
++#endif
+ }
+ 
+ /**
-- 
2.10.2

